tests/src/test/java/alluxio/client/fs/FileSystemReadonlyTest.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.client.fs;
+
+import static junit.framework.TestCase.assertTrue;
+
+import alluxio.AlluxioURI;
+import alluxio.client.file.FileSystem;
+import alluxio.client.file.FileSystemContext;
+import alluxio.conf.PropertyKey;
+import alluxio.conf.ServerConfiguration;
+import alluxio.exception.AccessControlException;
+import alluxio.master.file.FileSystemMaster;
+import alluxio.testutils.LocalAlluxioClusterResource;
+import alluxio.wire.MountPointInfo;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.util.Map;
+
+/**
+ * Test behavior of {@link FileSystemMaster}, when the paths are readonly.
+ */
+public class FileSystemReadonlyTest {
+  @Rule
```

Comment:
Can we make this a `ClassRule` so we dont have to re-create the cluster for each test? Starting a cluster is costly. (I think this also means the variable will have to be static)

Reply:
Done. BTW most integration tests are still using `@Rule`. We might want to batch-fix that in the future.

Reply:
Yes, I know. That is why I'm trying to make new usages use classrule. Thanks!